### PR TITLE
Return error if node/link name is too long

### PR DESCRIPTION
**Pull request by samhatchett**
co-authored with @ehsan-shafiee
... as reported in #534 

Return code will indicate "too many characters in input line" - which seems appropriate. Welcome other opinions.

### CODE REVIEW

File: src/input2.c [hunk 1]
```diff
@@ -472,7 +472,10 @@ int addnodeID(Network *net, int n, char *id)
 **--------------------------------------------------------------
 */
 {
-    if (findnode(net,id)) return 0;
+    if (findnode(net,id))
+      return 0;
+    if (strlen(id) >= MAXID)
+      return 214; //
```
Comment (eladsal) [inline src/input2.c]: the Slash signs can be removed :)

Comment (samhatchett) [reply src/input2.c]: you are right - I put them there thinking I would actually write a comment about the return code...

**Comment by LRossman** [comment]
There are two problems with these changes to the `addnodeID `and `addlinkID `functions:

1. Error 214 refers to having too many characters in an input file line, not too many characters in an ID name. A better error number to use is 252 "invalid ID name". (This number should also be used in input2.c's `addpattern `and `addcurve `functions instead of 250.)
2. Regardless of which error number is used, it never gets reported since the code that calls these functions (as in `juncdata`) only checks for a 0 return value, in which case it throws error 215.

A better way to address this issue is with the code I suggested in my issue #534 comment (but use error 252 instead of 250).

**Comment by samhatchett** [comment]
sounds like a good solution. Thanks @LRossman

**Comment by samhatchett** [comment]
I think this covers everything we discussed. Will someone else please review and merge?

**Comment by samhatchett** [comment]
I don't know why the one AppVeyor build has failed - anyone? Maybe related to the new benchmarks from #537 ?

**Comment by LRossman** [comment]
Yes, it's failing because of the new benchmarks for the `exnet-3` test files. Will re-fetching/updating your local branch from `dev `incorporate the #537 code changes into it, after which you can re-push to this PR?

**Comment by samhatchett** [comment]
aha! basic git stuff. Thanks @LRossman - we will see how the CI performs now.

**Pull request merged**